mlos_bench/mlos_bench/optimizers/convert_configspace.py

```diff
@@ -97,5 +130,91 @@ def tunable_values_to_configuration(tunables: TunableGroups) -> Configuration:
     Configuration
         A ConfigSpace Configuration.
     """
+    values: Dict[str, TunableValue] = {}
+    for (tunable, _group) in tunables:
+        if tunable.special:
+            (special_name, type_name) = special_param_names(tunable.name)
+            if tunable.value in tunable.special:
+                values[type_name] = "special"
+                values[special_name] = tunable.value
+            else:
+                values[type_name] = "range"
+                values[tunable.name] = tunable.value
+        else:
+            values[tunable.name] = tunable.value
     configspace = tunable_groups_to_configspace(tunables)
-    return Configuration(configspace, values={tunable.name: tunable.value for (tunable, _group) in tunables})
+    return Configuration(configspace, values=values)
+
+
+def configspace_data_to_tunable_values(data: dict) -> dict:
+    """
+    Remove the fields that correspond to special values in ConfigSpace.
+    In particular, remove `!type__` keys and trim `!special` suffixes.
```

Comment:
```suggestion
    In particular, remove and keys suffixes added by `special_param_names`.
```

Reword to avoid including the specific text for now? (in case it changes)
Maybe just reference the other routine where it was added.